packages/evm/contracts/payments/PaymentsBook.sol

```diff
@@ -0,0 +1,535 @@
+    * @return List of status to tell whether the corresponding token was claimed by the juror
+    */
+    function hasJurorClaimedMany(uint256 _periodId, address _juror, address[] calldata _tokens) external view returns (bool[] memory claimed) {
+        Period storage period = periods[_periodId];
+
+        claimed = new bool[](_tokens.length);
+        for (uint256 i = 0; i < _tokens.length; i++) {
+            claimed[i] = _hasClaimedJurorFees(period, _juror, _tokens[i]);
+        }
+    }
+
+    /**
+    * @dev Tell the fees corresponding to a juror for a certain period
+    * @param _periodId Identification number of the period being queried
+    * @param _token Address of the token to be queried
+    * @return Token amount corresponding to the governor
+    */
+    function getGovernorFees(uint256 _periodId, address _token) external view returns (uint256) {
+        Period storage period = periods[_periodId];
+        return period.governorFees[_token];
+    }
+
+    /**
+    * @dev Tell the fees corresponding to a juror for a certain period
+    * @param _periodId Identification number of the period being queried
+    * @param _tokens List of token addresses to be queried
+    * @return List of token amounts corresponding to the governor
+    */
+    function getManyGovernorFees(uint256 _periodId, address[] calldata _tokens) external view returns (uint256[] memory amounts) {
+        Period storage period = periods[_periodId];
+
+        amounts = new uint256[](_tokens.length);
+        for (uint256 i = 0; i < _tokens.length; i++) {
+            amounts[i] = _getGovernorFees(period, _tokens[i]);
+        }
+    }
+
+    /**
+    * @dev Internal function to claim juror fees for a certain period
+    * @param _period Period being claimed
+    * @param _periodId Identification number of the period claiming fees for
+    * @param _juror Address of the juror claiming the fees
+    * @param _token Address of the token being claimed
+    * @param _amount Amount of tokens to be transferred to the juror
+    */
+    function _claimJurorFees(Period storage _period, uint256 _periodId, address payable _juror, address _token, uint256 _amount) internal {
+        if (_amount > 0) {
+            _period.claimedJurorFees[_juror][_token] = true;
+            _transfer(_juror, _token, _amount);
+            emit JurorFeesClaimed(_periodId, _juror, _token, _amount);
+        }
+    }
+
+    /**
+    * @dev Internal function to transfer governor fees for a certain period
+    * @param _period Period being claimed
+    * @param _periodId Identification number of the period being claimed
+    * @param _token Address of the token to be claimed
+    */
+    function _transferGovernorFees(Period storage _period, uint256 _periodId, address payable _governor, address _token) internal {
+        uint256 amount = _getGovernorFees(_period, _token);
+        if (amount > 0) {
+            _period.governorFees[_token] = 0;
```

Comment:
While this works fabulously, a qualm I have about this is that it overwrites history for the governor's share in a way that does not happen for the guardian's share.

I'm wondering if we could repurpuse the `claimedGuardians` mapping such that it's also used for the governor?

Reply:
Addressed in 41d17890c03d2627e2baf44fd7c23843e9d10787